### PR TITLE
feat: add optional table_name attr

### LINKED ISSUE #195: feat: support configurable friendly names for tables
### TL;DR

Instead of defaulting to the `table_id`, I'd like to be able to optionally configure a user-friendly name for tables.

### Terraform Resources

[BigQuery Table friendly name](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_table#friendly_name)

### Detailed design

Update [this line](https://github.com/terraform-google-modules/terraform-google-bigquery/blob/master/main.tf#L70) to:

```hcl
friendly_name = each.value["table_name"] != null ? each.value["table_name"] : each.key
```

### PR DESCRIPTION
**Pull request by meghein**
# Summary

Resolves #195 - Adds support for optional `table_name` configuration.

# Test Plan

Confirm all tests pass and table's `friendly_name` variable will default to `table_id` if `table_name` is not configured.

### CODE REVIEW

**Comment by meghein** [comment]
Hi @bharathkkb and @apeabody 👋🏼  Just wondering if there is anything I can do to help in the review process or if there is someone I should reach out to instead? Thanks!

**Comment by apeabody** [comment]
Thanks for the contribution @meghein, triggering the CI.

/gcbrun

**Comment by meghein** [comment]
@apeabody - Any feedback on this? I'd love to help get this approved if there is anything additional that needs to be done. It looks like the CI passed originally. I've updated the branch as it has diverged from master - so I believe it will need to be triggered again.

**Comment by apeabody** [comment]
> @apeabody - Any feedback on this? I'd love to help get this approved if there is anything additional that needs to be done. It looks like the CI passed originally. I've updated the branch as it has diverged from master - so I believe it will need to be triggered again.

Thanks - I just triggered the CI.

File: README.md [hunk 1]
```diff
@@ -195,10 +196,10 @@ This module provisions a dataset and a list of tables with associated JSON schem
...
 | materialized\_views | A list of objects which includes view\_id, view\_query, clustering, time\_partitioning, range\_partitioning, expiration\_time and labels | <pre>list(object({<br>    view_id             = string,<br>    query               = string,<br>    enable_refresh      = bool,<br>    refresh_interval_ms = string,<br>    clustering          = list(string),<br>    time_partitioning = object({<br>      expiration_ms            = string,<br>      field                    = string,<br>      type                     = string,<br>      require_partition_filter = bool,<br>    }),<br>    range_partitioning = object({<br>      field = string,<br>      range = object({<br>        start    = string,<br>        end      = string,<br>        interval = string,<br>      }),<br>    }),<br>    expiration_time = string,<br>    labels          = map(string),<br>  }))</pre> | `[]` | no |
-| max\_time\_travel\_hours | Defines the time travel window in hours | `number` | `168` | no |
+| max\_time\_travel\_hours | Defines the time travel window in hours | `number` | `null` | no |
```
Comment (apeabody) [inline README.md:199]: Thanks - Looks like we had a stale value from https://github.com/terraform-google-modules/terraform-google-bigquery/pull/176

**Review by apeabody — APPROVED** [review]
Thanks for contribution!

It would be ideal to add the new `table_name` to an example for directly CI coverage, however `friendly_name` was previously being set to `each.key` so the new fall back behavior is already being tested by the CI.

**Pull request merged**